### PR TITLE
Support integration with real MCS controller

### PR DESCRIPTION
**Pull request by nmittler**
**Please provide a description of this PR:**

### CODE REVIEW

**Review by stevenctl — APPROVED** [review]

File: pilot/pkg/serviceregistry/kube/controller/serviceimportcache.go [hunk 1]
```diff
@@ -189,48 +179,78 @@ func (ic *serviceImportCacheImpl) onServiceImportEvent(obj interface{}, event mo
...
+					ips = append(ips, ip)
+				}
+			}
+		}
+	}
+	sort.Strings(ips)
+	return ips
+}
+
+// genMCSService generates an MCS service based on the given real k8s service. The llist of vips must be non-empty.
```
Comment (stevenctl) [inline pilot/pkg/serviceregistry/kube/controller/serviceimportcache.go:239]: typo: `llist`

File: pilot/pkg/serviceregistry/kube/controller/serviceimportcache.go [hunk 2]
```diff
@@ -189,48 +179,78 @@ func (ic *serviceImportCacheImpl) onServiceImportEvent(obj interface{}, event mo
...
-
-	if len(vips) > 0 {
-		mcsService.DefaultAddress = vips[0]
-		mcsService.ClusterVIPs.SetAddresses(map[cluster.ID][]string{
-			ic.Cluster(): vips,
-		})
-	} else {
-		mcsService.DefaultAddress = ""
-		mcsService.ClusterVIPs.SetAddresses(nil)
+	mcsService.DefaultAddress = vips[0]
```
Comment (stevenctl) [inline pilot/pkg/serviceregistry/kube/controller/serviceimportcache.go:243]: Is there a validator that would reject a CR with 0 IPs?

Comment (nmittler) [reply pilot/pkg/serviceregistry/kube/controller/serviceimportcache.go:243]: This is actually fine ... the assignment of the IP may happen after the ServiceImport is created.

**Pull request merged**